### PR TITLE
Tools: enable message parser and writer for different channels, including camera topics; fix lots of bugs

### PR DESCRIPTION
**Pull request by gchen-apollo**

### CODE REVIEW

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 1]
```diff
@@ -30,14 +30,14 @@
 import tarfile
 
 import six
-
+import numpy as np
 import cv2.cv as cv
-import pcl
+#import pcl
```
Comment (Capri2014) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: remove commented code

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 2]
```diff
@@ -72,15 +72,12 @@ def extract_camera_data(dest_dir, msg, ratio):
...
     # Check timestamp.
-    while True:
-        cur_time_second = msg.header.stamp.to_sec()
-        if cur_time_second - pre_time_second < ratio:
-            continue
-        pre_time_second = cur_time_second
-        time_nseconds.append(msg.header.stamp.to_nsec())
-
-        Image.ParseFromString(msg)
+    #while True:
```
Comment (Capri2014) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: same for this one

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 3]
```diff
@@ -72,15 +72,12 @@ def extract_camera_data(dest_dir, msg, ratio):
...
+    if True:
```
Comment (xiaoxq) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: Remove meaningless code, or add TODO if it's intended for future code.

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 4]
```diff
@@ -72,15 +72,12 @@ def extract_camera_data(dest_dir, msg, ratio):
...
+        cur_time_second = msg.timestamp
 
+        Image = sensor_image_pb2.Image()
```
Comment (xiaoxq) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: Name variables in camel case.

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py (hunk 1 above)
Comment (gchen-apollo) [reply modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: solved

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py (hunk 4 above)
Comment (gchen-apollo) [reply modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: solved

**Comment by freeHackOfJeff** [comment]
Looks good to me, many thanks!

**Pull request merged**